include/psp2kern/vfs.h

```diff
@@ -0,0 +1,810 @@
+        SceUID waitId;
+        SceUInt32 waitPattern;
+    };
+
+    SceUInt8 padding[44];
+
+    struct {  
+        SceVopTable *ops;
+        
+        SceUInt32 nodeInf; //!< Field for private VFS vnode info
+        void *nodeData;    //!< Pointer for private VFS data
+        
+        SceVfsMount *mnt;
+        
+        SceVfsVnode *dd;   //!<  Parent vnode
+        SceVfsVnode *next;
+        
+        SceUInt32 refCount;
+        
+        struct BuffCache *bc;
+
+        SceUInt32 fid[2];
+        struct Flock *flock;
+
+        SceUID allocator;
+        
+        struct Ncache *ncache;
+
+        SceUInt32 state; //!< One of SCE_VNODE_STATE_*
+        SceUInt32 type;  //!< ORed together SCE_VNODE_TYPE_* flags
+        
+        SceVopTable *vop_tbl; //!< Internal use only
+        
+        SceUInt64 size;
+        SceUInt32 aclData[2]; //!< Typically holds st_attr in aclData[0]
+
+        SceVfsFile *fdList;
+        SceUInt32 fdNum;
+        
+        SceVfsVnode *linkTo;
+        SceUInt32 linkedNum;
+        
+        SceUInt8 unused[48];
+        
+        SceUInt32 mntOpt;
+        SceUInt32 unk_0xd4;
+    };
+
+    SceUInt8 padding2[40];
+} SceVfsVnode;
+
+#define SCE_VFS_FILE_STATE_OPEN   (0x00000001) //!< File is open
+#define SCE_VFS_FILE_STATE_CLOSED (0x00000200) //!< File is closed
+#define SCE_VFS_FILE_STATE_UNK    (0x00000800) //!< Unknown status
+#define SCE_VFS_FILE_STATE_DUMMY  (0x00010000) //!< Dummy file
+
+/**
+ * VFS File
+ * 
+ * The file structure is contained within the SceUIDVfsFileObject structure, and represents file handles in the VFS.
+ */
+typedef struct SceVfsFile {
+    SceUInt32 isDir;
+    SceUInt32 flags;
+
+    SceOff position;
+
+    SceUInt32 state;
+    SceUID pid;
+    
+    SceVfsVnode *vp;
+    SceVfsFile *next;
+    
+    SceUInt32 fd;
+    SceUInt16 flockBusyCount;
+    
+    SceUInt8 isLocked;
+    SceUInt8 hasFlockEnt;
+    
+    struct FdLock *fdLock;
+    
+    SceUInt32 idata;
+    
+    struct DebugPath *debugPath; 
+    
+    SceUInt32 ioSchedData[3];
+} SceVfsFile;
+/** @} */
+
+/**
+ * @name VFS Operation Arguments and Callback Table
+ */
+/** @{ */
+typedef struct SceVfsOpMountArgs {
+	SceVfsMount *mnt;
+	SceVfsPath *devFilePath;
+} SceVfsOpMountArgs;
+
+typedef struct SceVfsOpUmountArgs {
+	SceVfsMount *mnt;
+	int flags;
+} SceVfsOpUmountArgs;
+
+typedef struct SceVfsOpSetRootArgs {
+	SceVfsMount *mnt;
+	int unk;
+	struct SceVfsVnode *vp;
+} SceVfsOpSetRootArgs;
+
+typedef struct SceVfsOpGetRootArgs {
+	SceVfsMount *mnt;
+	int unk;
+	struct SceVfsVnode **vpp;
+} SceVfsOpGetRootArgs;
+
+typedef struct SceVfsOpSyncArgs {
+	SceVfsMount *mnt;
+	int flags;
+} SceVfsOpSyncArgs;
+
+typedef struct SceVfsOpInitArgs {
+	SceVfsInfo *vfsInf;
+} SceVfsOpInitArgs;
+
+typedef struct SceVfsOpFiniArgs {
+    SceVfsInfo *vfsInf;
+} SceVfsOpFiniArgs;
+
+typedef struct SceVfsOpDevctlArg {
+	SceVfsMount *mnt;
+	const char *dev;
+	unsigned int cmd;
+    const void *arg;
```

Comment:
mixed indent type

i don't remember we use hardtab or softtab in future files.
@Princess-of-Sleeping do you remember this?

Reply:
I'll use hardtabs ('\t' characters) since sdstor.h, the most recently added header, does the same.